DataStructures/DoublyLinkedList/DoublyLinkedList.cs

```diff
@@ -0,0 +1,334 @@
+using System;
+using System.Collections.Generic;
+
+namespace DataStructures.DoublyLinkedList
+{
+    /// <summary>
+    /// Similar to a Singly Linked List but each node contains a refenrence to the previous node in the list.
+    /// <see cref="System.Collections.Generic.LinkedList{T}"/> is a doubly linked list.
+    ///
+    /// Compared to singly linked lists it can be traversed forwards and backwards.
+    /// Adding a node to a doubly linked list is simpler because ever node contains a reference to the previous node.
+    /// </summary>
+    /// <typeparam name="T">Generic type.</typeparam>
+    public class DoublyLinkedList<T>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DoublyLinkedList{T}"/> class.
+        /// </summary>
+        /// <param name="data"> Data of the original head of the list.</param>
+        public DoublyLinkedList(T data)
+        {
+            if (data is null)
```

Comment:
The build fails here with the following error.  "An expression of type 'T' cannot be handled by a pattern of type '<null>'. Please use language version 'preview' or greater to match an open type with a constant pattern." Basically, you can't match data with null because it may be a value type. The easiest fix is just to get rid of this check.